assets/js/app.js

```diff
@@ -49,4 +49,15 @@ var app = angular.module("Lucere", ["ngResource", "ngRoute", "dndLists"])
         templateUrl: "/js/templates/views/student_lesson.html"
       })
       .otherwise({redirectTo: "/"});
+    }])
+  .run(["$rootScope", "$location", "AuthService", function($rootScope, $location, AuthService) {
+    $rootScope.$on("$routeChangeStart", function(e) {
+      if(!AuthService.loggedIn()) {
```

Comment:
I'm hesitant because the asynchronous nature of this leaves some security vulnerability. However, for MVP it's fine. Just make sure you understand why this would be bad.